serviceaccounts.rb

```diff
@@ -18,20 +18,31 @@ def initialize(project, service_account = nil, path = nil)
     end
   end
 
+  def active_account()
+      return Common.new.capture_stdout %W{gcloud config get-value account}
+  end
+
   def run()
-    ENV["GOOGLE_APPLICATION_CREDENTIALS"] = @path
+    service_account = @service_account
+    if not service_account
+      service_account = "#{@project}@appspot.gserviceaccount.com"
+    end
+    if service_account == active_account()
+      # Don't generate another key if this account is already active. This can
+      # happen for nested service account contexts, for example.
```

Comment:
We could, but it would likely break separation of concerns. A given method requires some SA keyfile to exist, ideally it would work whether or not that SA key file already exists.

Either that or we plumb a bunch of extra parameters through to a bunch of places, which might be a larger job.